src/pages/ServerRoleSummary/UpdateNetworks.js

```diff
@@ -91,16 +110,55 @@ class UpdateNetworks extends Component {
       if(props.inputName === 'vlanid') {
         value = parseInt(value);
       }
+
       this.data[props.inputName] = value;
     }
   }
 
+  handleAddressChange = (e, isValid, props, idx) => {
+    let value = e.target.value;
+    if (isValid) {
+      if(!this.state.isFormChanged) {
+        this.setState({isFormChanged: true});
+      }
+      this.allAddressesStatus[idx] = INPUT_STATUS.VALID;
+
+      // this.data is the final result to be saved into model, init if we don't have any
+      // addresses will be processed when save later
+      if(!this.data['addresses']) {
+        this.data['addresses'] = [];
+      }
+    }
+    else {
+      this.allAddressesStatus[idx] = INPUT_STATUS.INVALID;
+    }
+
+    this.setState(prev => {
+      let addresses = prev.addresses.slice();
+      addresses[idx] = value;
+      return {addresses: addresses};
+    });
+    this.updateFormValidity(props, isValid);
+  }
+
   handleUpdateNetwork = () => {
     let model = this.props.model;
     for (let key in this.data) {
       if(this.data[key] === undefined || this.data[key] === '') {
         delete this.data[key];
       }
+
+      if (key === 'addresses') {
+        let cleanAddrs = this.state.addresses.filter((addr) => {
```

Comment:
This would be little cleaner and slightly less confusing to use:
``let cleanAddrs = this.state.addresses.filter(addr => addr !== '');``

Reply:
ok, will update